pkg/controller/cloud/node_controller.go

```diff
@@ -343,22 +343,24 @@ func excludeCloudTaint(taints []v1.Taint) []v1.Taint {
 
 // ensureNodeExistsByProviderID checks if the instance exists by the provider id,
 // If provider id in spec is empty it calls instanceId with node name to get provider id
-func ensureNodeExistsByProviderID(instances cloudprovider.Instances, node *v1.Node) (bool, error) {
+func ensureNodeExistsByProviderID(instances cloudprovider.Instances, node *v1.Node, cloud cloudprovider.Interface) (bool, error) {
 	providerID := node.Spec.ProviderID
 	if providerID == "" {
 		var err error
-		providerID, err = instances.InstanceID(context.TODO(), types.NodeName(node.Name))
```

Comment:
@andrewsykim Um...Sorry, I don't understand.

It looks support `<provider>://<instance-id>` format because bellow:

- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/aws/instances.go#L61-L72
- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/aws/instances.go#L52-L53

And other legacy cloud providers looks support `<provider>://<instance-id>` too:

**azure**

- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go#L43
- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go#L326-L330

**gcp**

- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/gce/gce_util.go#L206-L215

**vsphere**

- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere_util.go#L615-L619
- https://github.com/kubernetes/kubernetes/blob/41049fdf4b7b47742faf1f1265100f079afeccf6/staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere_util.go#L628-L630

Reply:
@andrewsykim Hi, as wrote above that looks like InstanceExistsByProviderID method in AWS and other providers support <provider>://<instance-id> format.
So maybe provider ID format could be unified.
But, shouldn't provider ID format be unified?

Reply:
Ideally it should, but there are some contraints here because we have to support legacy formats of the provider ID.

Reply:
Thank you. I understand.